### PR TITLE
fix: out of range access in 1D fftshifts

### LINKED ISSUE #151: Version mismatch
The `CMakeLists.txt` from release 0.2.0 still exposes version 0.1.0

### LINKED ISSUE #152: Errors in fftshift tests
Running the tests with debug features of Kokkos I got the following:
```
    Start 336: FFTShift/FFTShiftParamTests.Identity/9
1/2 Test #336: FFTShift/FFTShiftParamTests.Identity/9 ........Subprocess aborted***Exception:   0.10 sec
Note: Google Test filter = FFTShift/FFTShiftParamTests.Identity/0
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from FFTShift/FFTShiftParamTests
[ RUN      ] FFTShift/FFTShiftParamTests.Identity/0
Kokkos::View ERROR: out of bounds access label=("tmp") with indices [9] but extents [9]
Backtrace:
[0x7f4ba4cfc8e9] Kokkos::Impl::save_stacktrace()
[0x7f4ba4cd0860] Kokkos::Impl::host_abort(char const*)
[0x56058e9c9741] 
[0x56058ed3c14b] 
[0x56058ed3be14] 
[0x56058ed3baeb] 
[0x56058ed3b82b] 
[0x56058ed3b369] 
[0x56058ed19c80] 
[0x56058ed1436f] 
[0x7f4ba4dc2951] void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)
[0x7f4ba4da2e26] testing::Test::Run()
[0x7f4ba4da2fe5] testing::TestInfo::Run()
[0x7f4ba4da3295] testing::TestSuite::Run()
[0x7f4ba4db967c] testing::internal::UnitTestImpl::RunAllTests()
[0x7f4ba4d980ea] testing::UnitTest::Run()
[0x56058e9bb38d] 
[0x7f4ba4028083] __libc_start_main
[0x56058e990d5e] 

    Start 338: FFTShift/FFTShiftParamTests.1DShift1DView/9
2/2 Test #338: FFTShift/FFTShiftParamTests.1DShift1DView/9 ...Subprocess aborted***Exception:   0.09 sec
Note: Google Test filter = FFTShift/FFTShiftParamTests.1DShift1DView/0
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from FFTShift/FFTShiftParamTests
[ RUN      ] FFTShift/FFTShiftParamTests.1DShift1DView/0
Kokkos::View ERROR: out of bounds access label=("tmp") with indices [9] but extents [9]
Backtrace:
[0x7f5575ada8e9] Kokkos::Impl::save_stacktrace()
[0x7f5575aae860] Kokkos::Impl::host_abort(char const*)
[0x562caefd1741] 
[0x562caf34414b] 
[0x562caf343e14] 
[0x562caf343aeb] 
[0x562caf34382b] 
[0x562caf343369] 
[0x562caf321c80] 
[0x562caf31cbc2] 
[0x7f5575ba0951] void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)
[0x7f5575b80e26] testing::Test::Run()
[0x7f5575b80fe5] testing::TestInfo::Run()
[0x7f5575b81295] testing::TestSuite::Run()
[0x7f5575b9767c] testing::internal::UnitTestImpl::RunAllTests()
[0x7f5575b760ea] testing::UnitTest::Run()
[0x562caefc338d] 
[0x7f5574e06083] __libc_start_main
[0x562caef98d5e] 


0% tests passed, 2 tests failed out of 2

Total Test time (real) =   0.20 sec

The following tests FAILED:
	336 - FFTShift/FFTShiftParamTests.Identity/9 (Subprocess aborted)
	338 - FFTShift/FFTShiftParamTests.1DShift1DView/9 (Subprocess aborted)
Errors while running CTest
```

### PR DESCRIPTION
**Pull request by yasahi-hpc**
Fixes #152 
Fixes #151 

- [x] Add a range check inside a parallel for in `roll`
- [x] Build `OpenMP` backend with `Debug` in CI
- [ ] Update project version to `0.2.1`

### CODE REVIEW

File: .github/workflows/build_test.yaml [hunk 1]
```diff
@@ -76,6 +73,7 @@ jobs:
               kokkos: -DKokkos_ENABLE_SERIAL=ON
               kokkos_fft: -DCMAKE_CXX_FLAGS="-Wall -Wextra" -DCMAKE_COMPILE_WARNING_AS_ERROR=ON -DCMAKE_CXX_CLANG_TIDY="clang-tidy;-warnings-as-errors=*"
               benchmark: OFF
+            cmake_build_type: Debug
```
Comment (tpadioleau) [inline .github/workflows/build_test.yaml:76]: Is this flag also used to compile Kokkos ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,7 +9,7 @@ set(KOKKOSFFT_TOP_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 # Set project
 project(
     KokkosFFT
-    VERSION 0.1.0
+    VERSION 0.2.1
```
Comment (tpadioleau) [inline CMakeLists.txt:12]: I think we can make a dedicated PR ?

**Review by tpadioleau — APPROVED** [review]
LGTM, if I understand correctly it was a problem of implicit conversion signed/unsigned ?

File: CMakeLists.txt (hunk 2 above)
Comment (yasahi-hpc) [reply CMakeLists.txt:12]: Sure

**Comment by yasahi-hpc** [comment]
> LGTM, if I understand correctly it was a problem of implicit conversion signed/unsigned ?

Thank you for reviews.
That is a potential issue, so I have fixed. Current issue stems from a lack of a check function
```C++
if (i + shift0 < n0) {
  tmp(i + shift0) = inout(i);
}
```

File: .github/workflows/build_test.yaml (hunk 1 above)
Comment (yasahi-hpc) [reply .github/workflows/build_test.yaml:76]: Yes. Maybe keep using `Release` for `Kokkos`?

File: common/src/KokkosFFT_Helpers.hpp [hunk 3]
```diff
@@ -43,10 +43,10 @@ void roll(const ExecutionSpace& exec_space, ViewType& inout, axis_type<1> shift,
           axis_type<1>) {
   // Last parameter is ignored but present for keeping the interface consistent
   static_assert(ViewType::rank() == 1, "roll: Rank of View must be 1.");
-  std::size_t n0 = inout.extent(0);
+  int n0 = inout.extent(0);
```
[the remark this answers is not in the document]
Comment (yasahi-hpc) [reply common/src/KokkosFFT_Helpers.hpp:46]: Sure.  I will do that

File: .github/workflows/build_test.yaml (hunk 1 above)
Comment (tpadioleau) [reply .github/workflows/build_test.yaml:79]: No I think it is fine then. Debug enables bounds access in Kokkos views

**Review by tpadioleau — APPROVED** [review]
Ok for me!
